### PR TITLE
feat(protocol-designer): clear out engageHeight

### LINKED ISSUE #5225: PD: switching magnetic module model GEN1<->GEN2 clears engage heights
# Overview

The `engageHeight` field for GEN1 vs GEN2 magnetic modules should be in different units, to match behavior of Python Protocol API.

When changing the model GEN1 -> GEN2 or visa-versa in PD, we will clear out all `engageHeight` fields so that the user has to re-enter the numbers in new units

# Acceptance criteria

When you have a protocol with a magnetic module, changing its model should set all `engageHeight` fields to `null` for all Magnet forms, which should show the "required missing field" form error :warning:

### PR DESCRIPTION
**Pull request by IanLondon**
## overview

Clear out `engageHeight` from MAGNET steps when magnetic module model is
edited.

Closes #5225

I also cover an unspecified case where if you have a form open (thus in `unsavedForm` reducer state) and edit a module, that open form will just close without changes instead of doing something smarter

Also, I figured EDIT_MODULES shouldn't affect magnet steps that somehow aren't using the same `moduleId` as the edited module.

## changelog

* add tests for uncovered unsavedForm reducer

## review requests

- [ ] This is mostly code / test review b/c the UI to change module models isn't in yet
- [ ] One might argue that this form change should go thru `handleFormChange` as a best practice. I'm open to that, but IMO we should figure out #5247 first because this form isn't working right.
- [ ] You can test with redux devtools. You need to know the unique ID of your mag module, then dispatch:

```js
{
    type: 'EDIT_MODULE',
    payload: {
        id: "magnetic module ID here", 
        model: "magneticModuleV2"
    }
}
```

NOTE: you might notice that opening a Magnet > Disengage step shows "engage" being selected. That's not new in this PR, it's #5247 

## risk assessment

Definitely PD-only, shouldn't interfere with other PD work unless it's specifically engageHeight

### CODE REVIEW

File: protocol-designer/src/step-forms/test/reducers.test.js [hunk 1]
```diff
@@ -852,4 +878,145 @@ describe('savedStepForms reducer: initial deck setup step', () => {
...
+            moduleId: 'otherMagModuleId', // not 'magModuleId'
+            engageHeight: 12,
+          },
+        },
+      }
+      const result = savedStepForms(prevRootState, action)
+      expect(result).toEqual({
+        magnetEngageStepId: {
+          ...prevRootState.savedStepForms.magnetEngageStepId,
+          engage: true,
```
Comment (jen-fong) [inline protocol-designer/src/step-forms/test/reducers.test.js]: When I dispatch the action on react dev tools, the heights for engage steps clears out but the magnet action also seems to switch from engage to disengage. The test shows it's not supposed to happen, right? Is this related to the bug you mentioned?

Comment (IanLondon) [reply protocol-designer/src/step-forms/test/reducers.test.js]: that's probably the bug fixed in #5257
